src/app/services/xml-parsers/apparatus-entries-parser.service.ts

```diff
@@ -0,0 +1,131 @@
+import { Injectable } from '@angular/core';
+import { map, shareReplay } from 'rxjs/operators';
+import { AppConfig } from '../../../app/app.config';
+import { ApparatusEntriesList, ApparatusEntry, ApparatusReading, XMLElement } from '../../models/evt-models';
+import { xpath, getOuterHTML } from '../../utils/dom-utils';
+import { arrayToMap } from '../../utils/js-utils';
+import { removeSpaces } from '../../utils/xml-utils';
+import { EditionDataService } from '../edition-data.service';
+import { GenericParserService } from './generic-parser.service';
+import { WitnessesParserService } from './witnesses-parser.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ApparatusEntriesParserService {
+  public readonly appEntries$ = this.editionDataService.parsedEditionSource$.pipe(
+    map((source) => this.parseAppEntriesList(source)),
+    shareReplay(1),
+  );
+
+  private appEntryTagName = 'app';
+  private lemmaTagName = 'lem';
+  private readingTagName = 'rdg';
+  private readingGroupTagName = 'rdgGrp';
+  private noteTagName = 'note';
+  private significantRdgNumber = 0;
+  private parsedWitnesses = {};
+
+  constructor(
+    private editionDataService: EditionDataService,
+    private genericParserService: GenericParserService,
+    private witParserService: WitnessesParserService,
+  ) {
+  }
+
+  private parseAppEntriesList(document: XMLElement): ApparatusEntriesList {
+    const appEntries = Array.from(document.querySelectorAll<XMLElement>(this.appEntryTagName));
+
+    return this.parseAppEntries(appEntries);
+  }
```

Comment:
I think we should generalise this kind of function to be used everywhere in the parsers.
@ChiaraDipi

Reply:
What do you mean? Do you refer to a general parseList function?

Reply:
Yes, but maybe in a refactoring task